client/lib/user/utils.js

```diff
@@ -9,16 +9,21 @@ import debugModule from 'debug';
  */
 import config from 'config';
 import userModule from 'lib/user';
+import { once } from 'lib/memoize-last';
 
 /**
  * Module Variables
  */
-const user = userModule();
 const debug = debugModule( 'calypso:user:utilities' );
 
+// Delay initialisation of `user`, to avoid circular dependency issues.
+// `user` becomes a function that runs `userModule()` when it's first invoked,
+// saves the return value, and simply returns it on subsequent calls.
+const user = once( () => userModule() );
```

Comment:
The `userModule` function already does something like `once`: initializes a singleton variable and returns the existing instance after init.

It can be called very cheaply any number of times. In fact, we should probably replace all static, module-initialization-level calls to the `userFactory` with just-in-time invocations inside exported functions and classes.

Reply:
Good point, it does! I had been looking at the `user.js` file, but `index.js` wraps that in a singleton 👍

This library is calling out for a refactor 😕